### DISCUSSION TITLE
Direct access to s3 parquet files in Duckdb wasm

### DISCUSSION THREAD

**Discussion by kbatchu** [duckdb/duckdb-wasm#1718] [Q&A] [answered]
Hello,

I am able to access a parquet file hosted on s3 like this: select count(*) from read_parquet('https://data.source.coop/giswqs/nwi/wetlands/AK_Wetlands.parquet');

However, I get an error when I try to access the same file like this: select count(*) from read_parquet('s3://us-west-2.opendata.source.coop/giswqs/nwi/wetlands/AK_Wetlands.parquet'); The eror I get is: Invalid Error: NetworkError: Failed to execute 'send' on 'XMLHttpRequest': Failed to load 'https://us-west-2.opendata.source.coop.s3.amazonaws.com/giswqs/nwi/wetlands/AK_Wetlands.parquet'.

Does this mean that Duckdb wasm can access s3 files only through a custom domain connected to the s3 bucket?

PS: I loaded the spatial and httpfs extensions before I ran these commands.

Thanks,
Kiran

**Comment by Mause** [comment, marked as answer]
The error is due to a security certificate mismatch - if you try to open the first url, and allow chrome to bypass the error, it will then work

**Reply by kbatchu** [reply]
I am not sure I understood. Can you please explain in more detail?
Thanks,

**Comment by fgravin** [comment]
> PS: I loaded the spatial and httpfs extensions before I ran these commands.

A question out there, HTTPFS extension is not available in DuckDB WASM
See https://duckdb.org/docs/stable/clients/wasm/extensions.html#httpfs

Could you still load a geoparquet from S3 in WASM ?

When I try to read a geoparquet in WASM from `s3://` protocol, I have a `FAIL WITH: HEAD and GET requests failed: s3://overturemaps-us-west-2/release/2025-03-19.0/theme=places/type=place/*`

Do you think that I can load this file somehow ?
